src/common/perf_counters_collection.cc

```diff
@@ -0,0 +1,61 @@
+#include "common/perf_counters_collection.h"
+#include "common/ceph_mutex.h"
+#include "common/ceph_context.h"
+
+ /* PerfcounterCollection hold the lock for PerfCounterCollectionImp */
+PerfCountersCollection::PerfCountersCollection(CephContext *cct)
+  : m_cct(cct),
+    m_lock(ceph::make_mutex("PerfCountersCollection"))
+{
+  perf_impl = new PerfCountersCollectionImpl();
+}
+   PerfCountersCollection::~PerfCountersCollection()
+{
+  clear();
+}
+  void PerfCountersCollection:: add(class PerfCounters *l)
```

Comment:
please remove the leading spaces. and the space after `::`.

also `*` is part of the type, so i'd suggest put it right after `PerfCounters`.

Reply:
@tchaikov formatter is already set to json, and it can dump out the result.

Reply:
yes, will delete the space.